### DISCUSSION TITLE
The "i dati field is required", JSON correttamente formattato

### DISCUSSION THREAD

**Discussion by condor872** [fattureincloud/api#138] [Q&A] [answered]
Sto impazzendo nella creazione di una entry nel registro dei corrispettivi.

Questo è il codice:

$Token = MIO TOKEN;
$CompanyID = MIA COMPANY_ID;
$url= 'receipts';

$json_payload='{"data":{"date":"2022-10-29","numeration":"NEW-IT","amount_gross":-113.88,"use_gross_prices":true,"type":"sales_receipt","description":"TEST RICEVUTA","rc_center":"Italiano","payment_account":{"id":1484163,"name":"Contrassegno - Pagamento alla consegna"},"items_list":[{"amount_gross":-109.98,"vat":{"id":406009}},{"amount_gross":-3.19,"vat":{"id":406009}}]},"autocomplete_number":true}';

			$curlSend = curl_init();
			curl_setopt_array($curlSend, array(
			  CURLOPT_URL => 'https://api-v2.fattureincloud.it/c/'.$CompanyID.'/'.$url,
			  CURLOPT_RETURNTRANSFER => true,
			  CURLOPT_ENCODING => '',
			  CURLOPT_MAXREDIRS => 10,
			  CURLOPT_TIMEOUT => 0,
			  CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
			  CURLOPT_CUSTOMREQUEST => 'POST',
			  CURLOPT_POSTFIELDS =>$json_payload,
			  CURLOPT_HTTPHEADER => array(
				'Accept: application/json',
				'Authorization: Bearer '.$Token
			  ),
			));

			$response = curl_exec($curlSend);

			curl_close($curlSend);
			echo $response;

mi viene restituito questo messaggio
{"error":{"message":"Invalid request.","validation_result":{"data":["The i dati field is required."]}}}
Il json è correttamente formattato.
Ho anche provato a fare una get di una entry del registro dei corrispettivi, quindi copiare il json di risposta, modificando data, ed eliminado l'id della entry dal payload, e mi dice sempre lo stesso errore
ANCHE UTILIZZANDO POSTMAN, L'ERRORE RESTA LO STESSO

Utilizzando l'SDK , invece, in questo modo:

use FattureInCloud\Configuration as FicConfig;
use FattureInCloud\Api\ArchiveApi as FicArchiveApi;
use GuzzleHttp\Client as FicHTTPClient;
use GuzzleHttp\ClientInterface as FicHTTPClientInterface;
use FattureInCloud\Api\InfoApi as FicInfoApi;
use FattureInCloud\Api\IssuedDocumentsApi as FicIssuedDocumentsApi;
use FattureInCloud\Api\ReceiptsApi as FicReceiptsApi;
use FattureInCloud\Api\UserApi as FicUserApi;
use FattureInCloud\ApiException as FicApiException;

$config = FicConfig::getDefaultConfiguration()->setAccessToken($Token);
$apiInstance = new FicReceiptsApi(
			// If you want use custom http client, pass your client which implements `GuzzleHttp\ClientInterface`.
			// This is optional, `GuzzleHttp\Client` will be used as default.
			new FicHTTPClient(),
			$config
		);
//QUI I PARAMETRI LI PASSO COME ARRAY E NON COME UNA STRINGA JSON
$request=$apiInstance->CreateReceiptRequest($CompanyID, $params_array); 
$client = new FicHTTPClient();
try {
		$response=$client->send($request);
		$content = (string) $response->getBody();
		$response=json_decode($content,true);
	} 

catch (FicApiException $e) 
		{
			$response='Exception when calling: '.$e->getMessage(). 			PHP_EOL;
		}
Mi restituisce questo errore;

Client error: `POST https://api-v2.fattureincloud.it/c/MIA_COMPANY_ID/receipts` resulted in a `404 Not Found` response:  {"error":{"message":"Resource not found."}}
-CHIARAMENTE AL POSTO DI MIA_COMPANY_ID VIENE MOSTRATA IL VALORE RELATIVO-

**Comment by condor872** [comment]
![OnPaste 20221029-175728](https://user-images.githubusercontent.com/82886849/198841191-d844a445-f13e-4b1c-8e8d-65bb8a8f6ca1.png)
Vi mostro anche cosa vedo nel log dell'applicazione collegata. Come vedete, ancune richiesta hanno riposta 404, sono quelle fatte con l'SDK

**Comment by emajo** [comment, marked as answer]
Ciao @condor872 quei 404 sono dovuti probabilmente a qualche risorsa non presente sul tuo account tipo vat o payment account con id da te specificati, ti consiglio comunque di scrivere al nostro supporto clienti per andare a controllare i log delle tue chiamate e vedere nel dettaglio quale sia il problema.

Grazie,
Emanuele
